scripts/travis_doxygen.sh

```diff
@@ -0,0 +1,22 @@
+#!/bin/bash
+
+set -e
+
+script_folder=`dirname $0`
+pip install --user unidiff
+
+doxygen --version &> /dev/null
+if [[ $? -ne 0 ]]
+then
+  echo "doxygen not installed"
+  echo "Ensure doxygen is installed, then run again"
+  exit 1
+fi
+
+cd src && doxygen &> ../doxygen.log && cd ..
+
+if [ "$TRAVIS_PULL_REQUEST" == "false" ]; then
+  $script_folder/run_diff.sh 'cat doxygen.log' HEAD~1 # Check for errors introduced in last commit
```

Comment:
I would feel better about that, even though it might not win anyone a beauty contest. But this is about reliability and being understandable, not necessarily beauty,